overlord/devicestate/devicemgr.go

```diff
@@ -2214,3 +2214,93 @@ func (m *DeviceManager) RemoveRecoveryKeys() error {
 
 	return secbootRemoveRecoveryKeys(recoveryKeyDevices)
 }
+
+// EncryptionRequirements
+// XXX: move to secboot/encryption.go ?
+type EncryptionRequirements string
+
+const (
+	EncryptionRequired    EncryptionRequirements = "required"
+	EncryptionOptional    EncryptionRequirements = "optional"
+	EncryptionUnavailable EncryptionRequirements = "unavailable"
+	EncryptionDisabled    EncryptionRequirements = "disabled"
+)
+
+var secbootCheckTPMKeySealingSupported = secboot.CheckTPMKeySealingSupported
+
+// checkEncryption verifies whether encryption should be used based on the
+// model grade and the availability of a TPM device or a fde-setup hook
+// in the kernel.
+func (m *DeviceManager) checkEncryption(st *state.State, deviceCtx snapstate.DeviceContext) (res secboot.EncryptionType, err error) {
+	model := deviceCtx.Model()
+
+	kernelInfo, _ := snapstate.KernelInfo(st, deviceCtx)
+	// no stronger error check here because tests do not mock KernelInfo
+	// in enough places
+	_, res, err = m.checkEncryptionAndRequirements(model, kernelInfo)
+	return res, err
+}
+
+func (m *DeviceManager) checkEncryptionAndRequirements(model *asserts.Model, kernelInfo *snap.Info) (re EncryptionRequirements, encType secboot.EncryptionType, err error) {
```

Comment:
do you plan to call this one from the new API?

Reply:
Yes, the idea behind the refactor is that this could be called from the daemon and passed to the GET api for the installer.